scipy/sparse/_compressed.py

```diff
@@ -87,25 +86,24 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False, *, maxprint=None):
                 raise ValueError(f"unrecognized {self.__class__.__name__} "
                                  f"constructor input: {arg1}") from e
             if isinstance(self, sparray) and arg1.ndim < 2 and self.format == "csc":
-                raise ValueError(
-                    f"CSC arrays don't support {arg1.ndim}D input. Use 2D"
-                )
+                raise ValueError(f"CSC arrays don't support {arg1.ndim}D input. Use 2D")
             coo = self._coo_container(arg1, dtype=dtype)
             arrays = coo._coo_to_compressed(self._swap)
             self.indptr, self.indices, self.data, self._shape = arrays
 
         # Read matrix dimensions given, if any
         if shape is not None:
-            self._shape = check_shape(shape, allow_1d=is_array)
+            self._shape = check_shape(shape, allow_nd=self._allow_nd)
         elif self.shape is None:
             # shape not already set, try to infer dimensions
             try:
-                major_d = len(self.indptr) - 1
-                minor_d = self.indices.max() + 1
+                M = len(self.indptr) - 1
+                N = self.indices.max() + 1
             except Exception as e:
                 raise ValueError('unable to infer matrix dimensions') from e
 
-            self._shape = check_shape(self._swap((major_d, minor_d)), allow_1d=is_array)
+            self._shape = check_shape(self._swap((M, N)), allow_nd=self._allow_nd)
+            self._shape = check_shape(self._swap((M, N)), allow_nd=self._allow_nd)
```

Comment:
This duplication snuck in somehow.
```suggestion
            self._shape = check_shape(self._swap((M, N)), allow_nd=self._allow_nd)
```